### PR TITLE
Tidy up following I/O plugin migration

### PR DESCRIPTION
**Pull request by mppf**
Follow-on to PR #13327
  
* Skip HDFS and Curl testing except for in the test-networking-packages 
  test config since these require a service be started (so won't work if 
  tested in parallel by multiple developers, say)
* Re-document CHPL_AUX_FILESYS since it is used for lustre
* Remove some extern type/proc no longer available/used in IO.chpl
* Remove Makefile code specific to curl or hdfs in the runtime
* remove chplenv code specific to curl or hdfs in the runtime
* Remove third-party/libhdfs3 which is no longer available at the URL we 
  used to have for it and no longer used in the runtime
* Remove Curl dependency in modules/engin/pkg_module_res since it's 
  checking for compilation anyway

- [x] full local testing

Reviewed by @ronawho - thanks!

### CODE REVIEW

File: test/modules/engin/pkg_module_res.skipif [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# this test requires curl.h which isn't installed in nightly cygwin
```
Comment (ronawho) [inline test/modules/engin/pkg_module_res.skipif]: Is there a way to just check for the header instead of hardcoding that these machines lack it?

Comment (mppf) [reply test/modules/engin/pkg_module_res.skipif]: Yes but I wasn't sure it was worth the effort for this particular test, which used to be a compilation future. It doesn't actually use curl in a meaningful way and I also considered changing the test to not use curl at all.

Comment (ronawho) [reply test/modules/engin/pkg_module_res.skipif]: I'd be in favor of changing the test to not use curl.

Comment (mppf) [reply test/modules/engin/pkg_module_res.skipif]: As you wish!

**Pull request merged**